overlord/snapstate/snapmgr.go

```diff
@@ -556,6 +556,37 @@ func (m *SnapManager) RefreshSchedule() (string, bool, error) {
 	return m.autoRefresh.RefreshSchedule()
 }
 
+// EnsureAutoRefreshesAreDelayed will delay refreshes for the specified amount
+// of time, as well as return any active auto-refresh changes that are currently
+// not ready so that the client can wait for those.
+func (m *SnapManager) EnsureAutoRefreshesAreDelayed(delay time.Duration) ([]*state.Change, error) {
+
+	// TODO: what about gadget defaults for refresh.hold or other gadget refresh
```

Comment:
given that a gadget couldn't do something relative, this is very very unlikeky, I would simple drop this TODO